### ISSUE TITLE
[iOS] FontSubset error: Font subsetting failed with exit code 255

### ISSUE THREAD

**Issue by suztomo** [flutter/flutter#60325]
## Steps to Reproduce

0. Switch your flutter channel to beta (1.19.0-4.2.pre, which is the up-to-date beta)
1. Checkout https://github.com/suztomo/flutter-font-subsetting-failed. This main.dart contains the following code
```
void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        home: TextFormField(
      decoration: InputDecoration(
        // This fails: Target release_ios_bundle_flutter_assets failed:
        // FontSubset error: Font subsetting failed with exit code 255.
        icon: const Icon(CommunityMaterialIcons.pound_box),
        labelText: 'hello',
      ),
    ));
  }
}
```
2. Run the command below:
```
$ cd ios 
$ pod install 
$ flutter build ios      
   ```
3. You'll see the error `Target release_ios_bundle_flutter_assets failed: FontSubset error: Font subsetting failed with exit code 255.` (detail below)

**Expected results:** 

The build succeeds.

**Actual results:** 

The build fails with `Target release_ios_bundle_flutter_assets failed: FontSubset error: Font subsetting failed with exit code 255.` (detail below)

From my perspective, several things are broken in Flutter beta:
- I didn't specify "font subset", but somehow it tried to perform that operation.
- It could't perform "font subset" due to exit code 255 (???).
- The error message is obscure; it does not give me any indication on which file is wrong. I had to spend 1 day to identify the problematic piece of code is the input decoration with CommunityMaterialIcons.

# Workaround

See https://github.com/flutter/flutter/issues/60325#issuecomment-649900131

<details>
  <summary>Logs</summary>

```

Building suzotmo.hello for device (ios-release)...
Automatically signing iOS for device deployment using specified development team in Xcode project: M3QX44L9Y2
Running Xcode build...                                                  
                                                   
Xcode build done.                                           34.2s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    Failed to subset font; aborting.

    Target release_ios_bundle_flutter_assets failed: FontSubset error: Font subsetting failed with exit code
    255.
    build failed.
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
```

```
/tmp/font-subset-error/hello $ flutter analyze                                              git[branch:master]
Analyzing hello...                                                      
No issues found! (ran in 5.3s)
```

```
/tmp/font-subset-error/hello $ flutter doctor -v                                            git[branch:master]
[✓] Flutter (Channel beta, 1.19.0-4.2.pre, on Mac OS X 10.15.5 19F101, locale en-US)
    • Flutter version 1.19.0-4.2.pre at /Users/suztomo/development/flutter
    • Framework revision 9b9b543d92 (3 days ago), 2020-06-22 12:19:28 -0700
    • Engine revision 9a28c3bcf4
    • Dart version 2.9.0 (build 2.9.0-14.1.beta)

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, set ANDROID_SDK_ROOT to that location.
      You may also want to add it to your PATH environment variable.


 
[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.1

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] IntelliJ IDEA Community Edition (version 2020.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 46.0.3
    • Dart plugin version 201.7846.93

[✓] VS Code
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.8.1

 
[!] Connected device                          
    ! No devices available

! Doctor found issues in 3 categories.
```

</details>

**Comment by jonahwilliams** [comment]
FYI @dnfield 

@suztomo this can be worked around by passing `--no-tree-shake-icons` to `flutter build ios`

**Comment by dnfield** [comment]
Do the verbose logs give you a better error?

**Comment by suztomo** [comment]
@jonahwilliams It works. Thank you. (I'm now thinking how to apply it to my Fastlane setup)

@dnfield Thank you for response. That tells something is wrong about handling the community_material_icon.

https://gist.github.com/suztomo/b9d2737ea3b42090c5cc13b3f6c0e491

```
               [  +25 ms] Using output file:

/tmp/font-subset-error/hello/ios/Flutter/App.framework/flutter_assets/packages/community_material_icon/font
               s/materialdesignicons-webfont.ttf
                          Using source file:

/Users/suztomo/development/flutter/.pub-cache/hosted/pub.dartlang.org/community_material_icon-5.
                          3.45/lib/fonts/materialdesignicons-webfont.ttf
           
               [   +1 ms] Failed to subset font; aborting.
           
               [   +6 ms] Persisting file store
               [  +11 ms] Done persisting file store
               [   +9 ms] Target release_ios_bundle_flutter_assets failed: FontSubset error: Font subsetting
failed with
               exit code 255.
               [  +13 ms] "flutter assemble" took 13,964ms.
               build failed.
           
``

**Comment by suztomo** [comment]
I tried:

```
$ export TREE_SHAKE_ICONS=false
$ fastlane beta
```

But fastlane did not read the environment variable.

**Comment by jonahwilliams** [comment]
@suztomo I do not know how fastlane works, but you need to pass the flag to `flutter build ios`. We do not read any configuration from environment variables

**Comment by suztomo** [comment]
@jonahwilliams  I see TREE_SHAKE_ICONS variable in `$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh`. Isn't it supposed to control the behavior?

```
...
  RunCommand "${FLUTTER_ROOT}/bin/flutter"                                \
    ${verbose_flag}                                                       \
    ${flutter_engine_flag}                                                \
    ${local_engine_flag}                                                  \
    assemble                                                              \
    --output="${derived_dir}/"                                            \
    ${performance_measurement_option}                                     \
    -dTargetPlatform=ios                                                  \
    -dTargetFile="${target_path}"                                         \
    -dBuildMode=${build_mode}                                             \
    -dIosArchs="${ARCHS}"                                                 \
    -dSplitDebugInfo="${SPLIT_DEBUG_INFO}"                                \
    -dTreeShakeIcons="${TREE_SHAKE_ICONS}"                                \
    -dTrackWidgetCreation="${TRACK_WIDGET_CREATION}"                      \
    -dDartObfuscation="${DART_OBFUSCATION}"                               \
    -dEnableBitcode="${bitcode_flag}"                                     \
    --ExtraGenSnapshotOptions="${EXTRA_GEN_SNAPSHOT_OPTIONS}"             \
    --DartDefines="${DART_DEFINES}"                                       \
    -dExtraFrontEndOptions="${EXTRA_FRONT_END_OPTIONS}"                   \
    "${build_mode}_ios_bundle_flutter_assets"
...
```

Xcode invokes `$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh`.

<img width="1280" alt="Screen Shot 2020-06-25 at 9 12 44 PM" src="https://user-images.githubusercontent.com/28604/85810322-ae5b1100-b728-11ea-955f-c25fd40804da.png">


https://www.reddit.com/r/FlutterDev/comments/f4h3l9/tree_shake_icon_fonts_since_v1151/ also mentions the environment variable.

> If you're building from Xcode, you can set the TREE_SHAKE_ICONS environment variable


# How Fastlane works

- Fastlane invokes Xcode (xcodebuild)
- Xcode invokes `flutter_tools/bin/xcode_backend.sh`

**Comment by jonahwilliams** [comment]
You can certainly keep trying if you want to

**Comment by suztomo** [comment]
```
ENV["TREE_SHAKE_ICONS"] = "false"
```

didn't work either.


@jonahwilliams I think you're right; Flutter does not respect TREE_SHAKE_ICONS environmental variable any more. Xcode (and Fastlane) cannot turn the option off. I'll create another issue (unless this is intentional).

**Comment by jmagman** [comment]
Adding engine flags to make this error more actionable by the user:
https://github.com/flutter/engine/blob/abaac56c602cad3c1b3e41633bbfbe65200a1a3a/tools/font-subset/main.cc#L106

**Comment by jinfagang** [comment]
`flutter build apk` got simmilar error **only** when I try build apk, `flutter run` and `flutter run --release` all works fine.

```
Failed to subset font; aborting.



FAILURE: Build failed with an exception.

* Where:
Script 'E:\apps\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 896

* What went wrong:
Execution failed for task ':app:compileFlutterBuildRelease'.
> Process 'command 'E:\apps\flutter\bin\flutter.bat'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 54s
```
flutter is on latest channel dev.

**Comment by suztomo** [comment]
@jinfagang Which font failed?

`flutter build ios --verbose `

**Comment by awhitford** [comment]
This is my verbose output:
```
                         [ +747 ms] Running font-subset: /Users/anthony/flutter/bin/cache/artifacts/engine/darwin-x64/font-subset

/Users/anthony/github/kozzet/kozzeted/ios/Flutter/App.framework/flutter_assets/packages/flutter_vector_icons/fonts/Entypo.ttf
                         /Users/anthony/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_vector_icons-0.2.1/fonts/Entypo.ttf, using
codepoints 62032
                         [   +1 ms] Running font-subset: /Users/anthony/flutter/bin/cache/artifacts/engine/darwin-x64/font-subset

/Users/anthony/github/kozzet/kozzeted/ios/Flutter/App.framework/flutter_assets/packages/flutter_vector_icons/fonts/FontAwesome.ttf
                         /Users/anthony/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_vector_icons-0.2.1/fonts/FontAwesome.ttf, using
codepoints 62006
                         [        ] Running font-subset: /Users/anthony/flutter/bin/cache/artifacts/engine/darwin-x64/font-subset

/Users/anthony/github/kozzet/kozzeted/ios/Flutter/App.framework/flutter_assets/packages/flutter_vector_icons/fonts/Fontisto.ttf
                         /Users/anthony/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_vector_icons-0.2.1/fonts/Fontisto.ttf, using
codepoints 60203
                         [        ] Running font-subset: /Users/anthony/flutter/bin/cache/artifacts/engine/darwin-x64/font-subset

/Users/anthony/github/kozzet/kozzeted/ios/Flutter/App.framework/flutter_assets/packages/flutter_vector_icons/fonts/MaterialIcons.ttf
                         /Users/anthony/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_vector_icons-0.2.1/fonts/MaterialIcons.ttf, using
codepoints 58732
                         [        ] Running font-subset: /Users/anthony/flutter/bin/cache/artifacts/engine/darwin-x64/font-subset

/Users/anthony/github/kozzet/kozzeted/ios/Flutter/App.framework/flutter_assets/packages/flutter_vector_icons/fonts/MaterialCommunityIcon
                         s.ttf
/Users/anthony/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_vector_icons-0.2.1/fonts/MaterialCommunityIcons.ttf, using
                         codepoints 63888 61586 65041 63522 61926 983063 62179 64944 983066 62733 63104 61493 62125 61964 62788 63346 62275
983069 61676
                         [        ] Running font-subset: /Users/anthony/flutter/bin/cache/artifacts/engine/darwin-x64/font-subset

/Users/anthony/github/kozzet/kozzeted/ios/Flutter/App.framework/flutter_assets/packages/flutter_vector_icons/fonts/FontAwesome5_Solid.tt
                         f
/Users/anthony/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_vector_icons-0.2.1/fonts/FontAwesome5_Solid.ttf, using codepoints
                         62922
                         [   +1 ms] Running font-subset: /Users/anthony/flutter/bin/cache/artifacts/engine/darwin-x64/font-subset
                         /Users/anthony/github/kozzet/kozzeted/ios/Flutter/App.framework/flutter_assets/fonts/MaterialIcons-Regular.ttf
                         /Users/anthony/flutter/bin/cache/artifacts/material_fonts/MaterialIcons-Regular.ttf, using codepoints 58827 58828
59574 58834 58820
                         58848 58829 58825 58821 58831 58313 59701 58836 58835 58130 57746 57691 57669 59444 59445 57534 57344 59535 58826
57415 57398 57401
                         57395 59500 59511 60222 60224 60225 60227 59530 58715 60232 59651 60236 59601 58719 59412 59409 59410 59411 59413
57687 59540 57549
                         58917 57688 59624 59543 57777 57677 57542 59670 57403 57545 59389 58169 59369 59534 59527 57532 59542 57544 58714
59506 57535 59567
                         59613 58716 59529 59404 59569 59693 59618 58727 60233 57854 58669 58898 57671 59553 59568 57391 59513
                         [  +34 ms] Using output file:

/Users/anthony/github/kozzet/kozzeted/ios/Flutter/App.framework/flutter_assets/packages/flutter_vector_icons/fonts/MaterialCommunityIcon
                         s.ttf
                                    Using source file:

/Users/anthony/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_vector_icons-0.2.1/fonts/MaterialCommunityIcons.ttf
                     
                         [   +1 ms] Failed to subset font; aborting.
                     
                         [   +7 ms] Persisting file store
                         [  +11 ms] Done persisting file store
                         [  +10 ms] Target release_ios_bundle_flutter_assets failed: FontSubset error: Font subsetting failed with exit code
255.
                                    #0      IconTreeShaker.subsetFont
(package:flutter_tools/src/build_system/targets/icon_tree_shaker.dart:213:7)
                                    #1      _rootRunUnary (dart:async/zone.dart:1198:47)
                                    #2      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
                                    #3      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
                                    #4      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
                                    #5      Future._propagateToListeners (dart:async/future_impl.dart:725:32)
                                    #6      Future._complete (dart:async/future_impl.dart:519:7)
                                    #7      Stream.fold.<anonymous closure> (dart:async/stream.dart:847:14)
                                    #8      _rootRun (dart:async/zone.dart:1182:47)
                                    #9      _CustomZone.run (dart:async/zone.dart:1093:19)
                                    #10     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
                                    #11     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:410:13)
                                    #12     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:420:15)
                                    #13     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:305:7)
                                    #14     _SinkTransformerStreamSubscription._close (dart:async/stream_transformers.dart:97:11)
                                    #15     _EventSinkWrapper.close (dart:async/stream_transformers.dart:23:11)
                                    #16     _StringAdapterSink.close (dart:convert/string_conversion.dart:251:11)
                                    #17     _Utf8ConversionSink.close (dart:convert/string_conversion.dart:305:20)
                                    #18     _ConverterStreamEventSink.close (dart:convert/chunked_conversion.dart:82:18)
                                    #19     _SinkTransformerStreamSubscription._handleDone (dart:async/stream_transformers.dart:142:24)
                                    #20     _rootRun (dart:async/zone.dart:1182:47)
                                    #21     _CustomZone.run (dart:async/zone.dart:1093:19)
                                    #22     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
                                    #23     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:410:13)
                                    #24     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:420:15)
                                    #25     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:305:7)
                                    #26     _SyncStreamControllerDispatch._sendDone (dart:async/stream_controller.dart:816:19)
                                    #27     _StreamController._closeUnchecked (dart:async/stream_controller.dart:671:7)
                                    #28     _StreamController.close (dart:async/stream_controller.dart:664:5)
                                    #29     _Socket._onData (dart:io-patch/socket_patch.dart:2027:21)
                                    #30     _rootRunUnary (dart:async/zone.dart:1206:13)
                                    #31     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
                                    #32     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1005:7)
                                    #33     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:357:11)
                                    #34     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:285:7)
                                    #35     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:808:19)
                                    #36     _StreamController._add (dart:async/stream_controller.dart:682:7)
                                    #37     _StreamController.add (dart:async/stream_controller.dart:624:5)
                                    #38     new _RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1563:35)
                                    #39     _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:1045:18)
                                    #40     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
                                    #41     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
                                    #42     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
                                    #43     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
                     
                         [  +14 ms] "flutter assemble" took 60,848ms.
                         [   +7 ms] 
                                    #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
                                    #1      AssembleCommand.runCommand (package:flutter_tools/src/commands/assemble.dart:238:7)
                                    #2      _rootRunUnary (dart:async/zone.dart:1198:47)
                                    #3      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
                                    #4      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
                                    #5      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
                                    #6      Future._propagateToListeners (dart:async/future_impl.dart:725:32)
                                    #7      Future._completeWithValue (dart:async/future_impl.dart:529:5)
                                    #8      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
                                    #9      _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
                                    #10     FlutterBuildSystem.build (package:flutter_tools/src/build_system/build_system.dart)
                                    #11     _rootRunUnary (dart:async/zone.dart:1198:47)
                                    #12     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
                                    #13     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
                                    #14     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
                                    #15     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
                                    #16     Future._completeWithValue (dart:async/future_impl.dart:529:5)
                                    #17     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
                                    #18     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
                                    #19     _BuildInstance._invokeInternal (package:flutter_tools/src/build_system/build_system.dart)
                                    #20     _asyncErrorWrapperHelper.errorCallback (dart:async-patch/async_patch.dart:91:64)
                                    #21     _rootRunBinary (dart:async/zone.dart:1214:47)
                                    #22     _CustomZone.runBinary (dart:async/zone.dart:1107:19)
                                    #23     _FutureListener.handleError (dart:async/future_impl.dart:157:20)
                                    #24     Future._propagateToListeners.handleError (dart:async/future_impl.dart:708:47)
                                    #25     Future._propagateToListeners (dart:async/future_impl.dart:729:24)
                                    #26     Future._completeError (dart:async/future_impl.dart:537:5)
                                    #27     _AsyncAwaitCompleter.completeError (dart:async-patch/async_patch.dart:47:15)
                                    #28     IosAssetBundle.build (package:flutter_tools/src/build_system/targets/ios.dart)
                                    #29     _asyncErrorWrapperHelper.errorCallback (dart:async-patch/async_patch.dart:91:64)
                                    #30     _rootRunBinary (dart:async/zone.dart:1214:47)
                                    #31     _CustomZone.runBinary (dart:async/zone.dart:1107:19)
                                    #32     _FutureListener.handleError (dart:async/future_impl.dart:157:20)
                                    #33     Future._propagateToListeners.handleError (dart:async/future_impl.dart:708:47)
                                    #34     Future._propagateToListeners (dart:async/future_impl.dart:729:24)
                                    #35     Future._completeError (dart:async/future_impl.dart:537:5)
                                    #36     _AsyncAwaitCompleter.completeError (dart:async-patch/async_patch.dart:47:15)
                                    #37     copyAssets (package:flutter_tools/src/build_system/targets/assets.dart)
                                    #38     _asyncErrorWrapperHelper.errorCallback (dart:async-patch/async_patch.dart:91:64)
                                    #39     _rootRunBinary (dart:async/zone.dart:1214:47)
                                    #40     _CustomZone.runBinary (dart:async/zone.dart:1107:19)
                                    #41     _FutureListener.handleError (dart:async/future_impl.dart:157:20)
                                    #42     Future._propagateToListeners.handleError (dart:async/future_impl.dart:708:47)
                                    #43     Future._propagateToListeners (dart:async/future_impl.dart:729:24)
                                    #44     Future._completeError (dart:async/future_impl.dart:537:5)
                                    #45     Future.wait.<anonymous closure> (dart:async/future.dart:422:22)
                                    #46     _rootRunUnary (dart:async/zone.dart:1198:47)
                                    #47     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
                                    #48     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
                                    #49     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
                                    #50     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
                                    #51     Future._completeWithValue (dart:async/future_impl.dart:529:5)
                                    #52     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
                                    #53     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
                                    #54     copyAssets.<anonymous closure> (package:flutter_tools/src/build_system/targets/assets.dart)
                                    #55     _rootRunUnary (dart:async/zone.dart:1198:47)
                                    #56     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
                                    #57     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
                                    #58     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
                                    #59     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
                                    #60     Future._completeWithValue (dart:async/future_impl.dart:529:5)
                                    #61     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
                                    #62     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
                                    #63     IconTreeShaker.subsetFont (package:flutter_tools/src/build_system/targets/icon_tree_shaker.dart)
                                    #64     _rootRunUnary (dart:async/zone.dart:1198:47)
                                    #65     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
                                    #66     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
                                    #67     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
                                    #68     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
                                    #69     Future._completeWithValue (dart:async/future_impl.dart:529:5)
                                    #70     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
                                    #71     _rootRun (dart:async/zone.dart:1190:13)
                                    #72     _CustomZone.run (dart:async/zone.dart:1093:19)
                                    #73     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
                                    #74     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
                                    #75     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
                                    #76     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
                                    #77     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
                                    #78     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
                     
                     
                         [   +2 ms] ensureAnalyticsSent: 0ms
                         [   +2 ms] Running shutdown hooks
                         [        ] Shutdown hooks complete
                         [        ] exiting with code 1
                         Command PhaseScriptExecution failed with a nonzero exit code
```

**Comment by dnfield** [comment]
983069 looks suspicious as a codepoint in the community file.

**Comment by awhitford** [comment]
Does it really?  https://github.com/pd4d10/flutter-vector-icons/blob/master/lib/src/material_community_icons.dart#L103
```dart
  static const IconData account_edit_outline = IconData(983069,
      fontFamily: "MaterialCommunityIcons",
      fontPackage: "flutter_vector_icons");
```
Note that there other similar cases like [983068](https://github.com/pd4d10/flutter-vector-icons/blob/master/lib/src/material_community_icons.dart#L4) and [983067](https://github.com/pd4d10/flutter-vector-icons/blob/master/lib/src/material_community_icons.dart#L12253).

**Comment by andruli** [comment]
Getting this error in `stable` when building in release mode for ios or android

```
[   +9 ms] Target release_ios_bundle_flutter_assets failed: FontSubset error: Invalid ConstFinder result. Expected "fontPackage" to be a
String, "fontFamily"
                        to be a String, and "codePoint" to be an int, got: {codePoint: 32, fontFamily: null, fontPackage: null, matchTextDirection: false}.
                                   #0      IconTreeShaker._parseConstFinderResult (package:flutter_tools/src/build_system/targets/icon_tree_shaker.dart:307:9)
                                   #1      IconTreeShaker._findConstants (package:flutter_tools/src/build_system/targets/icon_tree_shaker.dart:298:12)
```

Using the `--no-tree-shake-icons` flag did the trick.

I need to supply the `verbose` flag in order to see what was going on, otherwise, the build process was just returning a `build failed` message without any extra information.

**Comment by dnfield** [comment]
Sorry, that codepoint should be fine. This is on my list right now, hoping to get to it this week.

**Comment by dnfield** [comment]
Harfbuzz doesn't seem to like this font. Not 100% sure why yet. For some reason, it's failing on the `hb_subset` call, as @jmagman pointed out in the link earlier.

We might want to just continue here - it may be better to subset fonts we can and ignore the ones we can't. But it'd be good to know why hb_subset is failing here.

**Comment by dnfield** [comment]
Ok. This does not fail on the latest version of harfbuzz.

@GaryQian was the last person to roll harfbuzz. Anything in particular to watch out for there?

I can work on prepping a roll for it.

**Comment by GaryQian** [comment]
I can roll harfbuzz again. Generally harfbuzz rolls are quite safe and usually result in strict improvements, the api is more or less completely frozen and non-breaking.

**Comment by GaryQian** [comment]
Some tests may shift a subpixel amount, but so far they have always looked better after than before

**Comment by dnfield** [comment]
Harfbuzz 2.7.0 introduced bugs for Khmer rendering and has to be reverted. 2.7.1 is available though and I'll try to roll to that asap.

**Comment by dnfield** [comment]
Fixed again in https://github.com/flutter/engine/pull/20609
